### DISCUSSION TITLE
Timeout connecting to server

### DISCUSSION THREAD

**Discussion by OMGAmici** [milvus-io/milvus#19775] [Q&A and General discussion]
Whenever I try to connect to the milvus server in the milvus cli using command `connect -h localhost -p 19530 -a default`, I receive ` (code=2, message=Fail connecting to server on localhost:19530. Timeout) `

I followed the documentation to launch Milvus through the docker compose file as standalone. 
![image](https://user-images.githubusercontent.com/83086025/195745192-2b74522d-8161-4d7e-a839-784465cd75d6.png)

When I test the connection using telnet and powershell tcp, it appears to be open and well (screenshot belows), so I really don't get what's happening here. 
![image](https://user-images.githubusercontent.com/83086025/195879014-b8b0231e-cf47-4cbe-8d8c-576bea86e0f2.png)
![image](https://user-images.githubusercontent.com/83086025/195880774-318d888f-79ba-46f8-9585-b1b881b51e70.png)

My research has shown that this error occurred to some other folks as well, but the fix was never clearly stated so I am hoping someone here can please guide me in the right direction.

**Comment by yhmo** [comment]
Possible reasons:
1. The service is not ready. 
2. Error occurs during starting the server. 

Check the log, if the server start successfully, you will see the two lines in log file:
```
[2022/10/20 18:25:58.085 +08:00] [DEBUG] [proxy/service.go:303] ["start Proxy server done"]
[2022/10/20 18:25:58.085 +08:00] [INFO] [components/proxy.go:57] ["Proxy successfully started"]
```

Your milvus version is 2.0.2, which is pretty old. Try the new version v2.1.4

In the next Milvus major version, we will provide an interface to check the healthy of Milvus server.

**Reply by OMGAmici** [reply]
Thank you!

I tried installed the more recent version, but the same error persists. 
![image](https://user-images.githubusercontent.com/83086025/197219171-ac40872d-2abb-4611-98e8-1a3798362330.png)

Steps I took:
1. Used Docker Stop to shut down Milvus 
2. `wget https://github.com/milvus-io/milvus/releases/download/v2.1.4/milvus-standalone-docker-compose.yml -O docker-compose.yml`
3. `sudo docker-compose up -d` to launch standalone Milvus
4. `docker run -it zilliz/milvus_cli:latest` to use the CLI
5. `connect -h localhost -p 19530 -a default` to test the connection from the CLI

I waited a few minutes after launching to ensure Milvus was ready and used docker ps to confirm that.
![image](https://user-images.githubusercontent.com/83086025/197219111-27c84b2a-f81d-47de-8418-790ae777b331.png)

Anything else I can try?

**Reply by OMGAmici** [reply]
If it's helpful, I am trying to set Milvus up in the context of using it with the Haystack NLP framework from DeepsetAI. They have a utility function launch_milvus() that I was using before. I can share more info on that note if it would be helpful.
